File: root/cgi-bin/get_username.py
#!/usr/bin/env python3
import os
import csv
import sys
import json

csv_path = os.path.join(os.path.dirname(__file__), "../../backend/users.csv")
session_id = os.environ.get("SESSION_ID", "")
username = ""

found = False  # flag
with open(csv_path, "r", newline='') as f:
    reader = csv.reader(f)
    for row in reader:
        if len(row) < 4:
            continue
        stored_username, stored_email, stored_password, stored_session_id = row
        if session_id == stored_session_id:
            username = stored_username
            found = True
            break

# Prepare JSON response
if found:
    response_data = {"username": username}
else:
    response_data = {"error": "Invalid session"}

response_json = json.dumps(response_data)

# Print headers and JSON without extra newlines or null bytes
sys.stdout.write("Content-Type: application/json\r\n\r\n")
sys.stdout.write(response_json)
sys.stdout.flush()

File: root/cgi-bin/get_username.php
#!/usr/bin/env php
<?php

$csv_path = __DIR__ . '/../../backend/users.csv';
$session_id = getenv("SESSION_ID") ?: "";
$username = "";
$found = false;

if (($handle = fopen($csv_path, "r")) !== false) {
    while (($row = fgetcsv($handle)) !== false) {
        if (count($row) < 4) {
            continue;
        }

        list($stored_username, $stored_email, $stored_password, $stored_session_id) = $row;

        if ($session_id === $stored_session_id) {
            $username = $stored_username;
            $found = true;
            break;
        }
    }
    fclose($handle);
}

// En-tête HTTP
header("Content-Type: application/json");

// Corps JSON
if ($found) {
    echo json_encode(["username" => $username]);
} else {
    echo json_encode(["error" => "Invalid session"]);
}
